lib/checkassert.cpp

```
@@ -129,6 +129,9 @@ void CheckAssert::checkVariableAssignment(const Token* assignTok, const Scope *a
     if (!assignTok->isAssignmentOp() && assignTok->tokType() != Token::eIncDecOp)
         return;
 
+    if (!assignTok->astOperand1())
```

I think `=` must always have both operands, so we should add this to the AST validation.

---

Unfortuantely, at least these cases would require special handling though:
~~~
auto f() { [=]() { return data; }(); }
~~~
~~~
class A {
    virtual void pure()=0;
};
~~~

---

So we leave it as is?

---

New idea: reject `=` followed by `;` and possibly `)|}` as garbage?

---

Do not ask me. There's a reason I usually do not get involved into the check related stuff...

---

IMHO we should fail early and hard on invalid code. Random nullptr checks here and there probably just enable the next crash down the line.

---

I split this into #6162.